libs/SalesforceSDK/src/com/salesforce/androidsdk/ui/PasscodeActivity.java

```diff
@@ -116,8 +114,11 @@ protected void onCreate(Bundle savedInstanceState) {
         if (shouldChangePasscode) {
             setMode(PasscodeMode.Change);
         } else {
-            setMode(passcodeManager.hasStoredPasscode(this) ? PasscodeMode.Check : PasscodeMode.Create);
-            showFingerprintDialog();
+            final PasscodeMode mode = passcodeManager.hasStoredPasscode(this) ? PasscodeMode.Check : PasscodeMode.Create;
+            setMode(mode);
+            if (mode == PasscodeMode.Check) {
```

Comment:
We want to ensure the `Create` and `Confirm` flows still happen, so that we have a passcode to fallback on.